network/hive_test.go

```diff
@@ -175,3 +185,228 @@ func TestHiveStatePersistence(t *testing.T) {
 		t.Fatalf("%d peers left over: %v", len(peers), peers)
 	}
 }
+
+/***
+ *
+ * - after connect, that outgoing subpeersmsg is sent
+ *
+ */
+func TestSubPeersMsg(t *testing.T) {
+	params := NewHiveParams()
+	s, pp, err := newHiveTester(params, 1, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	node := s.Nodes[0]
+	raddr := NewBzzAddrFromEnode(node)
+	pp.Register(raddr)
+
+	// start the hive and wait for the connection
+	pp.Start(s.Server)
+	defer pp.Stop()
+
+	// send subPeersMsg to the peer
+	err = s.TestExchanges(p2ptest.Exchange{
+		Label: "outgoing subPeersMsg",
+		Expects: []p2ptest.Expect{
+			{
+				Code: 1,
+				Msg:  &subPeersMsg{Depth: 0},
+				Peer: node.ID(),
+			},
+		},
+	})
+
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+const (
+	maxPO         = 8 // PO of pivot and control; chosen to test enough cases but not run too long
+	maxPeerPO     = 6 // pivot has no peers closer than this to the control peer
+	maxPeersPerPO = 3
+)
+
+// TestInitialPeersMsg tests if peersMsg response to incoming subPeersMsg is correct
+func TestInitialPeersMsg(t *testing.T) {
```

Comment:
It failed once https://ci.appveyor.com/project/ethersphere/swarm/builds/28070624/job/h8sjfqrhv9cnig19#L1526